modules/proto/core/tests/src/CompilerRendererSuite.scala

```diff
@@ -56,11 +99,15 @@ class CompilerRendererSuite extends FunSuite {
                       |
                       |message WithUnion {
                       |  int32 age = 1;
-                      |  oneof myUnion {
-                      |    string name = 2;
-                      |    int32 id = 3;
+                      |  com.example.MyUnion myUnion = 2;
```

Comment:
This changes the semantics completely of the interaction completely. We 100% need to retain the semantic of the union being inlined in the structure. One because we cannot just break our translation, now that people are using it, that'd be reckless and could cause bugs in production. 

Second because this change in encoding results in larger payloads at runtime. Not by much, but it's still larger, and when you use protobuf, it's very much to use as little payloads as possible.